File: 2563. Count the Number of Fair Pairs/countFairPairs.cpp
#include <iostream>
using namespace std;
#include <vector>

class Solution {
public:
    long long countPairs(vector<int>& nums,int target){
        long long cnt = 0;
        int left = 0;
        int right = nums.size()-1;

        while(left < right){
            int sum = nums[left] + nums[right];

            if(sum > target){
                right--;
            }
            else{
                cnt += (right-left);
                left++;
            }
        }
        return cnt;
    }
    long long countFairPairs(vector<int>& nums, int lower, int upper) {
        sort(nums.begin(),nums.end());

        return countPairs(nums, upper) - countPairs(nums, lower-1);
    }
};

File: 2563. Count the Number of Fair Pairs/countFairPairs.js

var countFairPairs = function (nums, lower, upper) {
    nums.sort((a, b) => a - b);

    function countPairs(target) {
        let left = 0;
        let right = nums.length - 1
        let cnt = 0
        while (left < right) {
            let sum = nums[left] + nums[right];
            if (sum > target) {
                right--
            }
            else {
                cnt += (right - left)
                left++
            }
        }
        return cnt;
    }
    return countPairs(upper) - countPairs(lower - 1)
};